src/main/java/org/prebid/server/bidder/ttx/TtxBidder.java

```diff
@@ -56,44 +60,68 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
 
     @Override
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
-        final Imp firstImp = request.getImp().get(0);
         final List<BidderError> errors = new ArrayList<>();
-        Imp updatedFirstImp = null;
-        ExtImpTtx extImpTtx = null;
-        try {
-            extImpTtx = parseImpExt(firstImp);
-            updatedFirstImp = updatedImp(firstImp, extImpTtx.getProductId(), extImpTtx.getZoneId(), errors);
-        } catch (PreBidException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-        }
-
-        if (updatedFirstImp != null && updatedFirstImp.getBanner() == null && updatedFirstImp.getVideo() == null) {
-            return Result.withError(BidderError.badInput("At least one of [banner, video] "
-                    + "formats must be defined in Imp. None found"));
-        }
+        final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
 
-        final HttpRequest<BidRequest> httpRequest = createRequest(request, extImpTtx, updatedFirstImp);
+        final ExtRequest reqExt = updateExtRequest(request.getExt());
 
-        return Result.of(Collections.singletonList(httpRequest), errors);
+        for (Imp imp : request.getImp()) {
+            try {
+                requests.add(makeRequest(request, imp, reqExt));
```

Comment:
makeRequest seems redundant. 
Could you split it like. 
validateImp()
parseImpExt()
updateImp()
 createRequest()